EnhancedMediaPlayer/MediaPlayer.swift

```diff
@@ -0,0 +1,92 @@
+import SwiftUI
+import AVKit
+
+struct MediaPlayerControllerRepresentable: UIViewControllerRepresentable {
+    private let mediaURL: URL
+    private let player = AVPlayer()
+    
+    init(mediaURL: URL) {
+        self.mediaURL = mediaURL
+    }
+    
+    func makeUIViewController(context: Context) -> AVPlayerViewController {
+        let playerController = AVPlayerViewController()
+        
+        playerController.showsPlaybackControls = false
+        playerController.player = player
+        playerController.player?.replaceCurrentItem(with: AVPlayerItem(url: mediaURL))
+        playerController.player?.play()
+
+        return playerController
+    }
+    
+    func updateUIViewController(_ uiViewController: AVPlayerViewController, context: Context) {
+        // No-op
+    }
+    
+    func play() {
+        player.play()
+    }
+    
+    func pause() {
+        player.pause()
+    }
+}
+
+public struct MediaPlayer: View {
+    @State private var isPlaying = true
+    @State private var showActionButtons = false
+    private let mediaURL: URL
+    private let playerController: MediaPlayerControllerRepresentable
+    
+    public init(url mediaURL: URL) {
+        self.mediaURL = mediaURL
+        self.playerController = MediaPlayerControllerRepresentable(mediaURL: mediaURL)
+    }
+    
+    public var body: some View {
+        ZStack {
+            playerController
+            
+            if showActionButtons {
+                renderActionButtons()
+            }
+        }
+        .onTapGesture { showActionButtons.toggle() }
+    }
+    
+    @ViewBuilder private func renderActionButtons() -> some View {
+        HStack {
+            renderPlayButton()
+        }
+    }
+
+    @ViewBuilder private func renderPlayButton() -> some View {
+        Button(action: playAndPause) {
+            Image(systemName: isPlaying ? Constants.playIcon : Constants.pauseIcon)
```

Comment:
This should be inverted, so that an user see's the play button when the video is paused
 ```suggestion
             Image(systemName: isPlaying ?  Constants.pauseIcon : Constants.playIcon)
```